### PR TITLE
Add support for Envoy 1.19.1

### PR DESCRIPTION
**Pull request by eculver**
#11100 - Adds support in our integration test suite and configuration for testing against Envoy 1.19.x. I'll drop support for 1.15 in a subsequent PR.

All tests are green, but I will go through and try to verify that aspects of what is new in 1.19.x are verified. For example, there was a default added for `connect_timeout`. Things like this still need to be verified as non-impacting. Also, there are a few [new features](https://www.envoyproxy.io/docs/envoy/v1.19.1/version_history/v1.19.0#new-features). I'm wondering if we need to do anything to address/validate them?

### CODE REVIEW

**Review by freddygv — COMMENTED** [review]
This is looking good, the only thing missing that stands out to me are the website updates:
https://www.consul.io/docs/connect/proxies/envoy#supported-versions
https://www.consul.io/commands/connect/envoy#envoy-version

Did you want to do that in the PR that removes 1.15?

**Comment by eculver** [comment]
@freddygv actually, let me see if removing support for 1.15 works here. I was trying to keep it simple, but given there were no significant changes for 1.19, I can try to remove 1.15 and see how it goes.

**Comment by eculver** [comment]
> @freddygv actually, let me see if removing support for 1.15 works here. I was trying to keep it simple, but given there were no significant changes for 1.19, I can try to remove 1.15 and see how it goes.

Scratch that. I just realized that there are some feature flags that are only relevant for 1.15 so will require some more surgery than I originally thought.

What are your thoughts on having two separate changelog entries? One for adding 1.19 and the other for removing 1.15? And then separate website updates to match?

**Review by freddygv — APPROVED** [review]
That approach sounds good @eculver. 

Can you also update the default version listed here?
https://www.consul.io/commands/connect/envoy#envoy-version

File: command/connect/envoy/testdata/defaults.golden [hunk 1]
```diff
@@ -14,7 +14,7 @@
     "metadata": {
       "namespace": "default",
       "partition": "default",
-      "envoy_version": "1.18.4"
+      "envoy_version": "1.19.1"
```
Comment (rboyer) [inline command/connect/envoy/testdata/defaults.golden:17]: @eculver maybe afterwards you could knock this one out too: https://github.com/hashicorp/consul/issues/11134

**Comment by rboyer** [comment]
> http: the behavior of the `present_match` in route header matcher changed. The value of `present_match` was ignored in the past. The new behavior is `present_match` is performed when the value is true. An absent match performed when the value is false. Please reference `present_match`.

We use this for L7 routing and also L7 RBAC

**Comment by rboyer** [comment]
> tcp: switched to the new connection pool by default. Any unexpected behavioral changes can be reverted by setting runtime guard envoy.reloadable_features.new_tcp_connection_pool to false.

:eyes:

**Comment by rboyer** [comment]
> listener: fix a crash which could happen when a filter chain only listener update is followed by listener removal or a full listener update.
yay!

**Comment by eculver** [comment]
@rboyer did you ever make anything of this:
> tcp: switched to the new connection pool by default. Any unexpected behavioral changes can be reverted by setting runtime guard envoy.reloadable_features.new_tcp_connection_pool to false.

Otherwise, I think this is good to merge.

**Comment by rboyer** [comment]
@eculver 
> tcp: switched to the new connection pool by default. Any unexpected behavioral changes can be reverted by setting runtime guard envoy.reloadable_features.new_tcp_connection_pool to false.

Nothing for us here, mostly just keeping an eye out for this refactor should something weird happen.

**Review by rboyer — APPROVED** [review]
LGTM

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/463488.

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/465809.

**Comment by hc-github-team-consul-core** [comment]
:cherries::x: Cherry pick of commit c7747212c3f0735c521508abe94749cc5d188b2a onto `stable-website` failed! [Build Log](https://circleci.com/gh/hashicorp/consul/465809)

**Comment by hc-github-team-consul-core** [comment]
:cherries::x: Cherry pick of commit c7747212c3f0735c521508abe94749cc5d188b2a onto `release/1.10.x` failed! [Build Log](https://circleci.com/gh/hashicorp/consul/465809)

**Pull request merged**